### PR TITLE
Add tenant activate and deactivate by tenant id

### PR DESCRIPTION
**Pull request by dewniMW**

### CODE REVIEW

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java [hunk 1]
```diff
@@ -209,6 +209,33 @@ public void activateTenant(String tenantUniqueID) throws TenantMgtException {
...
+                        tenantDomain, PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername()));
+            } else {
+                throw new TenantManagementClientException(ERROR_CODE_RESOURCE_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), tenantId));
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new TenantManagementServerException(String.format("Error while retrieving or deactivating the " +
+                    "tenant with id: %s", tenantId), e);
+        } catch (StratosException e) {
+            throw new TenantManagementServerException(String.format("Error in notifying tenant activation of tenant " +
```
Comment (wenodpathirana) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:234]: `tenantDomain` could be null if `activateTenant()` fails.

Comment (dewniMW) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:234]: activateTenant() throws a UserStoreException. StratosException is thrown from triggerTenantActivation method. hence tenantDomain will not be null.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java [hunk 2]
```diff
@@ -209,6 +209,33 @@ public void activateTenant(String tenantUniqueID) throws TenantMgtException {
...
+                tenantManager.activateTenant(tenantId);
+                tenantDomain = tenant.getDomain();
+                // Notify tenant activation to all listeners.
+                TenantMgtUtil.triggerTenantActivation(tenantId);
+                log.info(String.format("Tenant with id: %s and domain: %s activated by %s", tenantId,
+                        tenantDomain, PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername()));
+            } else {
+                throw new TenantManagementClientException(ERROR_CODE_RESOURCE_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), tenantId));
+            }
```
Comment (AnuradhaSK) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:229]:
```suggestion
            if (tenant == null) {
                 throw new TenantManagementClientException(ERROR_CODE_RESOURCE_NOT_FOUND.getCode(),
                        String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), tenantId));
            } 
            tenantManager.activateTenant(tenantId);
                tenantDomain = tenant.getDomain();
                // Notify tenant activation to all listeners.
                TenantMgtUtil.triggerTenantActivation(tenantId);
                log.info(String.format("Tenant with id: %s and domain: %s activated by %s", tenantId,
                        tenantDomain, PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername()));
```

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java [hunk 3]
```diff
@@ -239,6 +266,34 @@ public void deactivateTenant(String tenantUniqueID) throws TenantMgtException {
...
+                // Notify tenant deactivation to all listeners.
+                TenantMgtUtil.triggerTenantDeactivation(tenantId);
+                tenantManager.deactivateTenant(tenantId);
+                TenantMgtUtil.unloadTenantConfigurations(tenantDomain, tenantId);
+                log.info(String.format("Tenant with id: %s and domain: %s deactivated by %s", tenantId,
+                        tenantDomain, PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername()));
+            } else {
+                throw new TenantManagementClientException(ERROR_CODE_RESOURCE_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), tenantId));
+            }
```
Comment (AnuradhaSK) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:287]: Can do similar thing as https://github.com/wso2/carbon-multitenancy/pull/222/files#r871979742

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java (hunk 2 above)
Comment (dewniMW) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java]: addressed

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**